module/module.go

```diff
@@ -172,18 +175,21 @@ func (m *Module) Start(ctx context.Context) error {
 
 // Close shuts down the module and grpc server.
 func (m *Module) Close(ctx context.Context) {
-	m.mu.Lock()
-	defer m.mu.Unlock()
-	m.logger.Info("Shutting down gracefully.")
-	if m.parent != nil {
-		if err := m.parent.Close(ctx); err != nil {
+	m.closeOnce.Do(func() {
+		m.mu.Lock()
```

Comment:
in windows we were shutting this process down too quickly because of how process groping and CtrlC works there (changed that in goutils now so this doesn't get hit). this makes sure we don't hold the lock and cause a race where RemoveResource is happening while Closing (happened a lot in Windows)